observer.cpp

```diff
@@ -163,6 +165,57 @@ void test_observer()
     }
 #endif
 
+#define FIX_ISSUE_271
+#ifdef FIX_ISSUE_271
+    ///The observer stores performance calculated from reaction norm
+    {
+        observer o;
+
+        //Give sim some non-default inputs and optimal
+        simulation s{};
+        o.store_performance(s);
+        assert(o.get_average_perf()[0] == calc_avg_perf_from_reaction_norm(s));
+        assert(o.get_sd_perf()[0] == calc_sd_perf_from_reaction_norm(s));
+    }
+#endif
 
+#define FIX_ISSUE_272
+#ifdef FIX_ISSUE_272
+    ///Average and variance of performance can be obtained from a reaction norm
+    /// Note: this assumes one output
```

Comment:
I think we can make this more general by making the optimal value a vector of doubles as well, and by defining an operation that calculates the pairwise distance of the elements of 2 vectors (ouput of the reaction norm and optimal output)